#!/usr/bin/env python
PACKAGE = "gpadteleop"

import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

from gpadteleop.msg import GamePadLevels

gen = ParameterGenerator()

gen.add("max_lin_vel",double_t,GamePadLevels.GPADLEVEL_MAXLIN,"Maximum Linear Velcoity of Robot",1.0,0,30)
gen.add("max_ang_vel",double_t,GamePadLevels.GPADLEVEL_MAXANG,"Maximum Angular Velocity of Robot",0.5,0,10)
gen.add("percent_lin_slow",double_t,GamePadLevels.GPADLEVEL_MAXLIN,"Slow Mode Linear Speed of Robot in Percentage of Maximum Speed",0.2,0,1.0)
gen.add("percent_lin_medium",double_t,GamePadLevels.GPADLEVEL_MAXLIN,"Medium Mode Linear Speed of Robot in Percentage of Maximum Speed",0.6,0,1.0)
gen.add("percent_lin_fast",double_t,GamePadLevels.GPADLEVEL_MAXLIN,"Fast Mode Linear Speed of Robot in Percentage of Maximum Speed",1.0,0,1.0)
gen.add("percent_lin_inc",double_t,GamePadLevels.GPADLEVEL_MAXLIN,"Percentage Increment of Linear Speed in Fixed Speed Mode",0.1,0,0.5)
gen.add("percent_ang_inc",double_t,GamePadLevels.GPADLEVEL_MAXANG,"Percentage Increment of Angular Speed in Fixed Speed Mode",0.1,0,0.5)
gen.add("percent_ang_slow",double_t,GamePadLevels.GPADLEVEL_MAXANG,"Slow Mode Angular Speed of Robot in Percentage of Maximum Speed",0.2,0,1.0)
gen.add("percent_ang_medium",double_t,GamePadLevels.GPADLEVEL_MAXANG,"Medium Mode Angular Speed of Robot in Percentage of Maximum Speed",0.6,0,1.0)
gen.add("percent_ang_fast",double_t,GamePadLevels.GPADLEVEL_MAXANG,"Fast Mode Angular Speed of Robot in Percentage of Maximum Speed",1.0,0,1.0)
gen.add("axis_linear",int_t,GamePadLevels.GPADLEVEL_AXIS_LIN,"GamePad Axis ID for Linear Motion",1,0,6)
gen.add("axis_angular",int_t,GamePadLevels.GPADLEVEL_AXIS_ANG,"GamePad Axis ID for Angular Motion",3,0,6)
gen.add("axis_linear_speed",int_t,GamePadLevels.GPADLEVEL_AXIS_LINSPEED,"GamePad Axis ID for Linear Motion Speed",6,0,6)
gen.add("axis_angular_speed",int_t,GamePadLevels.GPADLEVEL_AXIS_ANGSPEED,"GamePad Axis ID for Angular Motion Speed",5,0,6)
gen.add("btn_lin_mode",int_t,GamePadLevels.GPADLEVEL_BTN_LINMODE,"GamePad Button ID for Linear Speed Mode Selection",6,0,11)
gen.add("btn_ang_mode",int_t,GamePadLevels.GPADLEVEL_BTN_ANGMODE,"GamePad Button ID for Angular Speed Mode Selection",7,0,11)
gen.add("btn_hard_brake",int_t,GamePadLevels.GPADLEVEL_BTN_HRDBRK,"GamePad Button ID for Hard Brake which also disables motors",4,0,11)
gen.add("btn_soft_brake",int_t,GamePadLevels.GPADLEVEL_BTN_SFTBRK,"GamePad Button ID for Soft Brake",5,0,11)
gen.add("btn_robot_lock",int_t,GamePadLevels.GPADLEVEL_BTN_LOCK,"GamePad Button ID for engaging/disengaging of lock",3,0,11)
gen.add("btn_speed_mode_lin",int_t,GamePadLevels.GPADLEVEL_BTN_TGLLIN,"GamePad Button ID for toggling Speed Mode of Linear Motion",10,0,11)
gen.add("btn_speed_mode_ang",int_t,GamePadLevels.GPADLEVEL_BTN_TGLANG,"GamePad Button ID for toggling Speed Mode of Angular Motion",11,0,11)
gen.add("publish_freq",int_t,GamePadLevels.GPADLEVEL_PUBFREQ,"Publish Frequency of Commands in Speed Mode in Hz",30,1,120)
gen.add("srv_motors_enable",str_t,GamePadLevels.GPADLEVEL_MOTORE,"Service to Enable Motors","enable_motors")
gen.add("srv_motors_disable",str_t,GamePadLevels.GPADLEVEL_MOTORD,"Service to Disable Motors","disable_motors")

exit(gen.generate(PACKAGE,"GPadTeleop2","GamePad"))
